### PR TITLE
Two improvements of sync_diff_inspector

### PR DESCRIPTION
**Pull request by qixiaobin**
### What problem does this PR solve? 
When using sync_diff_inspector, if I have a table in my target DB but not in source DB, the process will be terminated after report a "check failed" message. This is not quite convenient. Because sometimes we just create a few tables in our target DB for work and those tables are not in source DB.
So I make two changes in sync_diff_inspector:
1.When a target table not in source DB, just log it and go on (instead of terminating the process).
2.Add an "exclude_tables" part in "check-tables" section in the config.toml

```toml
[[check-tables]]
    # schema name in target database.
    schema = "mydb"

    # table list which need check in target database.
    #tables = ["task","user"]

    # support regular expression, must start with '~'.
    tables = ["~^"]
    
    # table list which need NOT to be checked in target database.
    exclude_tables = ["a_table"]
```

### What is changed and how it works?
1.Modify the diff.go to continue to work when target table not exist in source DB.
2.Modify config.go to add "ExcludeTables".
3.Modify diff.go to exclude tables in "AdjustTableConfig" function.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
1.Create table in target DB
2. Test to see if the process is going on after report a "check failed" message.
3. Add that table into "exclude_tables" in config.toml and re-run sync_diff_inspector to see the result.

Code changes

 - Has exported function/method change
 - Has exported variable/fields change

Side effects

 - Possible performance regression
 - Increased code complexity

Related changes
 - Need to update the documentation

### CODE REVIEW

**Comment by WangXiangUSTC** [comment]
@qixiaobin thanks, we will take a look

File: sync_diff_inspector/diff.go
Comment (WangXiangUSTC) [inline sync_diff_inspector/diff.go]: please delete this line

File: sync_diff_inspector/diff.go
Comment (WangXiangUSTC) [inline sync_diff_inspector/diff.go]: please delete this line

File: sync_diff_inspector/diff.go
Comment (WangXiangUSTC) [inline sync_diff_inspector/diff.go]: please rename `r` to a meaningful name

**Review by WangXiangUSTC — COMMENTED** [review]
rest LGTM

File: sync_diff_inspector/diff.go
Comment (qixiaobin) [reply sync_diff_inspector/diff.go]: thanks. 
finished. 
pls check.

**Comment by WangXiangUSTC** [comment]
LGTM, and It would be better if we could add unit test or integration test 😃

**Comment by qixiaobin** [comment]
> LGTM, and It would be better if we could add unit test or integration test 😃

thx. So how can this be approving reviewed and get merged? I think those two issues are quite often needed when comparing DB structures.

**Comment by WangXiangUSTC** [comment]
> > LGTM, and It would be better if we could add unit test or integration test 😃
> 
> thx. So how can this be approving reviewed and get merged? I think those two issues are quite often needed when comparing DB structures.

Need another LGTM, please wait

File: sync_diff_inspector/config.go
Comment (csuzhangxc) [inline sync_diff_inspector/config.go]: we often use `-` in field tags, so how about renaming `exclude_tables` to `exclude-tables`? If so, do not forget to update the one in _config.toml_.

File: sync_diff_inspector/diff.go
Comment (csuzhangxc) [inline sync_diff_inspector/diff.go:451]: Should we need to support regex as we did for `Tables`?

**Review by csuzhangxc — COMMENTED** [review]
rest LGTM

File: sync_diff_inspector/config.go
Comment (qixiaobin) [reply sync_diff_inspector/config.go]: sure, I'll change it. thanks.

File: sync_diff_inspector/diff.go
Comment (qixiaobin) [reply sync_diff_inspector/diff.go:451]: Well, for me, we just occasionally create a few tables in target-db that aren't  in source-db and they don't have common prefix/postfix. So maybe we can just make it simple.
Like to hear other's advices.

File: sync_diff_inspector/config.go
Comment (qixiaobin) [reply sync_diff_inspector/config.go]: Finished.

**Comment by qixiaobin** [comment]
> LGTM, and It would be better if we could add unit test or integration test 😃

Good idea. I'll do it when I have time.

File: sync_diff_inspector/diff.go
Comment (csuzhangxc) [reply sync_diff_inspector/diff.go:451]: OK, you can keep it simple now.

**Review by csuzhangxc — APPROVED** [review]
LGTM, and thanks a lot.

**Comment by qixiaobin** [comment]
My pleasure.发自我的华为手机-------- 原始邮件 --------发件人： Xuecheng Zhang <notifications@github.com>日期： 2020年5月29日周五 22:39收件人： pingcap/tidb-tools <tidb-tools@noreply.github.com>抄送： qixiaobin <chopinqi@163.com>, Mention <mention@noreply.github.com>主    题： Re: [pingcap/tidb-tools] Two improvements of sync_diff_inspector (#346)
@csuzhangxc approved this pull request.

LGTM, and thanks a lot.

**Comment by qixiaobin** [comment]
> LGTM, and It would be better if we could add unit test or integration test 😃

Test added and I create another pull request for it.

**Comment by qixiaobin** [comment]
I’ve add test for the ‘exclude-tables’ config and create a pull request for it. Pls check.

Sent from Mail for Windows 10

File: sync_diff_inspector/diff.go
Comment (WangXiangUSTC) [inline sync_diff_inspector/diff.go:432]: shoud save the check result like `df.report.FailedNum++`, fix in 
https://github.com/pingcap/tidb-tools/pull/366

**Pull request merged**
